src/ess/reduce/ui.py

```diff
@@ -232,3 +234,80 @@ def refresh_workflow_box(change) -> None:
     workflow_selection_box = widgets.HBox([workflow_select], layout=default_layout)
     workflow_box = widgets.Box(layout=default_layout)
     return widgets.VBox([workflow_selection_box, workflow_box])
+
+
+def _get_parameter_box(widget: WorkflowWidget | ParameterBox) -> ParameterBox:
+    if isinstance(widget, WorkflowWidget):
+        return widget.parameter_box
+    elif isinstance(widget, ParameterBox):
+        return widget
+    else:
+        raise TypeError(
+            f"Expected target_widget to be a WorkflowWidget or ParameterBox, "
+            f"got {type(widget)}."
+        )
+
+
+def update_parameter_widget(
+    widget: WorkflowWidget | ParameterBox, new_parameter_values: dict[type, Any]
+) -> None:
+    """Set the values of the input widgets in the target widget.
+
+    Nodes that don't exist in the input widgets will be ignored.
+
+    Example
+    -------
+    {
+        'WavelengthBins': {'start': 1.0, 'stop': 14.0, 'nbins': 500}
+    }
+
+    Parameters
+    ----------
+    widget:
+        The widget containing the input widgets.
+    new_parameter_values:
+        A dictionary of values/state to set each fields/state or value of input widgets.
+
+    Raises
+    ------
+    TypeError:
+        If the widget is not a WorkflowWidget or a ParameterBox.
+
+    """
+    parameter_box = _get_parameter_box(widget)
+    # Walk through the existing input widgets and set the values
+    # ``node`s that don't exist in the input widgets will be ignored.
+    for node, widget in parameter_box._input_widgets.items():
+        if node in new_parameter_values:
+            # We shouldn't use `get` here because ``None`` is a valid value.
+            set_fields(widget, new_parameter_values[node])
+
+
+def get_parameter_widget_values(
```

Comment:
Should we try to find "symmetric" naming? Would be easier to remember and figure out that the output of one is the input for the other. We had `update_parameter_widget` above.